pkg/testing/framework/utils.go

```diff
@@ -95,6 +97,23 @@ func FromEnvTestConfig(cfg *rest.Config) api.Config {
 	return c
 }
 
+// GetClusterMinorVersion returns Kubernetes server minor version (only digits)
+func GetClusterMinorVersion(kc kubernetes.Interface) (int, error) {
+	info, err := kc.Discovery().ServerVersion()
+	if err != nil {
+		return 0, err
+	}
+	minor := strings.TrimRightFunc(strings.Split(info.Minor, ".")[0], func(r rune) bool {
+		return !unicode.IsDigit(r)
+	})
```

Comment:
```suggestion
	minor := info.Minor[:strings.indexFunc(info.Minor, func(r rune) bool {
		return !unicode.IsDigit(r)
	})]
```

what about this?

I'm trying to make it safer as I'm not sure of what values we can get there now:

- 31+rc1 should currently fail if I'm not wrong.